### PR TITLE
Add Mandatory CICD Template to Project (--cicd)

### PR DESCRIPTION
**Pull request by jacobevansgit**
- If ADO is the desired git repo, azure-pipelines.yml folder will be created in the root
- If Github is the desired git repo, .github/workflows will be the path for the python-package.yml
- Default is github

### CODE REVIEW

File: src/templates/shared/cicd/azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+trigger:
+- master
+
+pool:
+  vmImage: ubuntu-latest
+strategy:
+  matrix:
+    Python27:
+      python.version: '2.7'
```
Comment (asbfall) [inline src/templates/shared/cicd/azure-pipelines.yml:14]: I would not put python 2.x

File: src/templates/shared/cicd/python-package.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
```
Comment (asbfall) [inline src/templates/shared/cicd/python-package.yml:23]: it is not redundant with what the .flake8 file should contain?

**Review by asbfall — COMMENTED** [review]
I do not see the github/workflow files

File: src/templates/shared/cicd/python-package.yml (hunk 2 above)
Comment (jacobevansgit) [reply src/templates/shared/cicd/python-package.yml]: Removed the cli flags in favour of .flake settings

File: src/generator.py [hunk 3]
```diff
@@ -137,7 +145,19 @@ def generate_template_at_destination(self, destination: pathlib.Path) -> None:
         ProjectTemplateAndDestinationChecker(self.template, destination).check()
         copy_directory_and_contents_to_new_location(self.template.template_directory, destination)
         if self.template.shared_files_directory is not None:
-            copy_directory_and_contents_to_new_location(self.template.shared_files_directory, destination)
+            copy_directory_and_contents_to_new_location(self.template.shared_files_directory,
+                                                        destination,
+                                                        just_files=True)
+            if self.template.cicd_type == 'github':
+                cicd_target = destination / '.github' / 'workflows'
```
Comment (jacobevansgit) [inline src/generator.py:152]: @asbfall  Regarding the .github / workflow files, we are copying the file to this folder if the 
`--cicd 'github'` flag is applied (Default)

**Review by jgd10 — APPROVED** [review]
LGTM!

**Pull request merged**